src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -648,7 +648,13 @@ private string GetDebuggerDisplay()
                 {
                     var variableOpt = variablesOpt?[i].Single;
                     uint valEscape = variableOpt is null ? LocalScopeDepth : GetValEscape(variableOpt, LocalScopeDepth);
-                    var variable = new OutDeconstructVarPendingInference(receiverSyntax, valEscape);
+                    var variableSymbol = variableOpt switch
+                    {
+                        DeconstructionVariablePendingInference { VariableSymbol: var symbol } => symbol,
+                        BoundLocal { DeclarationKind: BoundLocalDeclarationKind.WithExplicitType or BoundLocalDeclarationKind.WithInferredType, LocalSymbol: var symbol } => symbol,
```

Comment:
We have other cases where we check `DeclarationKind: not BoundLocalDeclarationKind.None`. In future we might want to be consistent here, or maybe introduce an extension/computed property which asks if the bound-local represents a user-visible variable declaration, which throws when an unknown DeclarationKind is encountered.

No need to change anything in this PR.